pkg/serviceproxy/impersonation_test.go

```
@@ -0,0 +1,566 @@
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := extractClientImpersonation(test.headers)
+			if test.wantStatus == 0 {
+				if err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+				if !reflect.DeepEqual(got, test.want) {
+					t.Fatalf("unexpected request:\nwant: %#v\ngot:  %#v", test.want, got)
+				}
+			} else {
+				assertRequestErrorStatus(t, err, test.wantStatus)
```

golangci-lint (2.12.2)

[error] Error return value is not checked

(errcheck)